plugins/BEdita/API/src/Auth/EndpointAuthorize.php

```diff
@@ -146,31 +159,36 @@ public function isAnonymous($user)
 
     /**
      * Get application for request.
+     * This is done primarily with an API_KEY header like 'X-Api-Key',
+     * alternatively `api_key` query string is used (not recommended)
      *
      * @return \BEdita\Core\Model\Entity\Application|null
      * @throws \Cake\Network\Exception\ForbiddenException Throws an exception if API key is missing or invalid.
      */
     protected function getApplication()
     {
-        $application = CurrentApplication::getApplication();
-        if ($application === null) {
-            $header = $this->request->getHeaderLine($this->_config['apiKeyHeaderName']);
-            if (empty($header) && empty($this->_config['blockAnonymousApps'])) {
+        $this->application = CurrentApplication::getApplication();
+        if ($this->application === null) {
+            $apiKey = $this->request->getHeaderLine($this->_config['apiKeyHeaderName']);
+            if (empty($apiKey)) {
+                $apiKey = strval($this->request->getQuery($this->_config['apiKeyQueryString']));
```

Comment:
(minor) I generally prefer type-casting, thus `(string)$var` instead of `strval($var)`. They should yield the same results, so it's just a matter of taste.